cmd/mt-kafka-mdm-sniff/main.go

```diff
@@ -0,0 +1,96 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"math/rand"
+	"os"
+	"os/signal"
+	"strconv"
+	"syscall"
+	"text/template"
+
+	inKafkaMdm "github.com/raintank/metrictank/input/kafkamdm"
+	"github.com/raintank/metrictank/stats"
+	"github.com/raintank/worldping-api/pkg/log"
+	"github.com/rakyll/globalconf"
+	"gopkg.in/raintank/schema.v1"
+)
+
+var (
+	confFile = flag.String("config", "/etc/raintank/metrictank.ini", "configuration file path")
+	format   = flag.String("format", "{{.Part}} {{.OrgId}} {{.Id}} {{.Name}} {{.Metric}} {{.Interval}} {{.Value}} {{.Time}} {{.Unit}} {{.Mtype}} {{.Tags}}", "template to render the data with")
+)
+
+type Data struct {
+	Part int32
+	schema.MetricData
+}
+
+type inputPrinter struct {
+	template.Template
+	data Data
+}
+
+func newInputPrinter(format string) inputPrinter {
+	tpl := template.Must(template.New("format").Parse(format + "\n"))
+	return inputPrinter{
+		*tpl,
+		Data{},
+	}
+}
+
+func (ip inputPrinter) Process(metric *schema.MetricData, partition int32) {
+	ip.data.MetricData = *metric
+	ip.data.Part = partition
+	err := ip.Execute(os.Stdout, ip.data)
+	if err != nil {
+		log.Error(0, "executing template: %s", err)
+	}
+}
+
+func main() {
+	flag.Usage = func() {
+		fmt.Fprintln(os.Stderr, "mt-kafka-mdm-sniff")
+		fmt.Fprintln(os.Stderr)
+		fmt.Fprintln(os.Stderr, "Inspects what's flowing through kafka (in mdm format) and reports it to you")
+		fmt.Fprintf(os.Stderr, "\nFlags:\n\n")
+		flag.PrintDefaults()
+	}
+	flag.Parse()
+	log.NewLogger(0, "console", fmt.Sprintf(`{"level": %d, "formatting":false}`, 2))
+	instance := "mt-kafka-mdm-sniff" + strconv.Itoa(rand.Int())
+
+	// Only try and parse the conf file if it exists
+	path := ""
+	if _, err := os.Stat(*confFile); err == nil {
```

Comment:
no. `NewWithOptions` doesn't tried to load a file if the filename is "".

BTW which editor do you use? I use neovim with vim-go. There's a really neat key-bind `gd` (go to definition), so that when you're in our source code, and you're on `NewWithOptions` and you want to know how it works, you can just hit gd and it takes you to the source code of it (and ctrl-o to go back). this is really neat for cases like this.